src/utilities/transformAPIdata.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  IChapter,
+  IComicChapters,
+  ISingularChapterMeta,
+} from "../models/chapterData";
+import { IComic, IComicData } from "../models/comicData";
+import { ICarouselComic, TopComics } from "../models/homePageData";
+import { IMangaImage } from "../models/mangaImageData";
+
+export function getHomePageData(
+  comics: TopComics,
+  sectionName: string
+): ICarouselComic[] {
+  switch (sectionName) {
+    case "trending":
+      const trendingData: ICarouselComic[] = comics?.trending[30];
+      trendingData.filter((comic) => comic.content_rating === "safe");
```

Comment:
You could maybe make this a function as you reuse it a lot.

Reply:
That would be tricky in typescript cause I filter different types of arrays each time... i think it would force me to use any if i extract that filtering into a function